crates/ty_python_semantic/src/types/call/arguments.rs

```diff
@@ -172,7 +175,16 @@ impl<'a, 'db> CallArguments<'a, 'db> {
                 index += 1;
             };
 
-            let mut expanded_arguments = Vec::with_capacity(expanded_types.len() * previous.len());
+            let expansion_size = expanded_types.len() * previous.len();
+            if expansion_size > MAX_EXPANSIONS {
+                tracing::debug!(
+                    "Skipping argument type expansion as it would exceed the \
```

Comment:
Sorry for commenting on a draft PR. Is there some information that we could include to make it easier for users to narrow down where this is happening (e.g. the full signature of the called function?, or some information about the call site?)